### ISSUE TITLE
Flutter tool bin path could be duplicate in Setting up the Framework development environment?

### ISSUE THREAD

**Issue by feinstein** [flutter/flutter#37282]
This isn't a bug exactly, it's just that I am looking at how to contribute to Flutter and the `Setting up the Framework development environment` page says:

`7. Add this repository's bin directory to your path. That will let you use the flutter command in this directory more easily.`

Wouldn't this make a conflict in my machine, as I already have the `flutter` tool installed (the one that comes in the regular installation of Flutter)?

**Comment by miyoyo** [comment]
PATH works similarily in most OSes, which is, the first match always wins.
In fact, you probably already have many such conflicts without you doing anything special, it's just the first one always primes over the others.
Here's an example:
```
[miyoyo@Alzheimer ~]$ whereis java
java: /usr/bin/java /usr/share/java /usr/lib/jvm/java-8-openjdk/bin/java
```
in this case, I have 3 java executables on my path, however, only the first one (/usr/bin/java) will be used when I invoke `java` in a terminal.

**Comment by feinstein** [comment]
I know, but still, won't this be an issue with the `flutter` tool?

I have seen it complains when I use the `flutter` tool from one repository into another repository, so I will always have one complaining at me, be it when I am developing Flutter itself, or an app using the stable installation I have.

Maybe the docs should give the tip to create an alias to the development repository `flutter` tool? This way we could use both without warnings?

**Comment by miyoyo** [comment]
The flutter tool isn't per-repository, the complaints it has are related to the machine-specific folder references stored in metadata that shouldn't be uploaded anyway.

Aliases aren't really an options, they are not supported in Windows' CMD.exe at all, and anything referring to the `flutter` executable *will still refer to the original tool*, which is likely to cause errors.

**Comment by feinstein** [comment]
So the recommendation from the dev team is to "replace" (path-wise) the **original tool** with the **development** one? IIRC the flutter tool will detect a different one was last used and automatically switch repositories, won't this impact development of new apps? 

Maybe a solution would be to make a shortcut for the development flutter tool, just to have a different name, like `flutterdev` and thus having it on the path won't bring any ambiguities?

**Comment by feinstein** [comment]
How do you guys develop the Flutter Framework and Flutter apps at the same time? Do you just ignore the warnings from the flutter tool like:

>Warning! The 'flutter' tool you are currently running is from a different Flutter repository than the one last used by
this package. The repository from which the 'flutter' tool is currently executing will be used instead.
  running Flutter tool: C:\Programacao\Flutter\sdk\flutter
  previous reference  : C:\Programacao\Flutter\flutter
This can happen when you have multiple copies of flutter installed. Please check your system path to verify that you are
running the expected version (run 'flutter --version' to see which flutter is on your path).

**Comment by jonahwilliams** [comment]
If you're developing the flutter framework, you need to use the flutter tool from that framework. That is, the flutter sdk is only guaranteed to work with a single flutter tool, the one that it ships with

**Comment by feinstein** [comment]
Yes, I am aware, I am just concerned on the best way to setup my machine (and probably update the wiki's recommendations) for both app and framework development, thus having 2 flutter tools installed on the path, and be able to use both without any problems.
